tests/test_opds_for_distributors.py

```diff
@@ -0,0 +1,358 @@
+
+    def test_patron_activity(self):
+        # The patron has two loans from this API's collection and
+        # one from a different collection.
+        patron = self._patron()
+
+        data_source = DataSource.lookup(self._db, "Biblioboard", autocreate=True)
+        e1, p1 = self._edition(
+            identifier_type=Identifier.URI,
+            data_source_name=data_source.name,
+            with_license_pool=True,
+            collection=self.collection,
+        )
+        p1.loan_to(patron)
+
+        e2, p2 = self._edition(
+            identifier_type=Identifier.URI,
+            data_source_name=data_source.name,
+            with_license_pool=True,
+            collection=self.collection,
+        )
+        p2.loan_to(patron)
+
+        other_collection = self._collection(protocol=ExternalIntegration.OVERDRIVE)
+        e3, p3 = self._edition(
+            identifier_type=Identifier.OVERDRIVE_ID,
+            data_source_name=DataSource.OVERDRIVE,
+            with_license_pool=True,
+            collection=other_collection,
+        )
+        p3.loan_to(patron)
+        
+        activity = self.api.patron_activity(patron, "1234")
+        eq_(2, len(activity))
+        [l1, l2] = activity
+        eq_(l1.collection_id, self.collection.id)
+        eq_(l2.collection_id, self.collection.id)
+        eq_(set([l1.identifier, l2.identifier]),
+            set([p1.identifier.identifier, p2.identifier.identifier]))
+
+class TestOPDSForDistributorsImporter(DatabaseTest, BaseOPDSForDistributorsTest):
+
+    def test_import(self):
+        feed = self.get_data("biblioboard_mini_feed.opds")
+
+        data_source = DataSource.lookup(self._db, "Biblioboard", autocreate=True)
+        collection = MockOPDSForDistributorsAPI.mock_collection(self._db)
+        collection.external_integration.set_setting(
+            Collection.DATA_SOURCE_NAME_SETTING,
+            data_source.name
+        )
+
+        class MockMetadataClient(object):
+            def canonicalize_author_name(self, identifier, working_display_name):
+                return working_display_name
+        importer = OPDSForDistributorsImporter(
+            self._db, collection=collection,
+            metadata_client=MockMetadataClient(),
+        )
+
+        imported_editions, imported_pools, imported_works, failures = (
+            importer.import_from_feed(feed)
+        )
+
+        # This importer works the same as the base OPDSImporter, except that
+        # it adds delivery mechanisms for books with epub acquisition links
+        # and sets pools' licenses_owned and licenses_available.
+
+        # Both works were created, since we can use their acquisition links
+        # to give copies to patrons.
+        [camelot, southern] = sorted(imported_works, key=lambda x: x.title)
+
+        # Each work has a license pool.
+        [camelot_pool] = camelot.license_pools
+        [southern_pool] = southern.license_pools
```

Comment:
It would be good to see what happens with a book that has an OPDS entry but no acquisition link. Not that important since that's not how Biblioboard gates a library's access to books.